### PR TITLE
feat: cairo 1

### PR DESCRIPTION
**Pull request by tserg**

### CODE REVIEW

**Review by bllu404 — APPROVED** [review]
LGTM!

File: README.md
Comment (milancermak) [inline README.md:5]: StarkNet -> Starknet

File: README.md
Comment (milancermak) [inline README.md:42]: Can you add a note somewhere around here about the loss of precision with some of these operations? Just to be really obvious about it.

File: README.md
Comment (milancermak) [inline README.md:72]: Obsolete now.

File: README.md
Comment (milancermak) [inline README.md:78]: "by implementing" sounds a little like the users still need to implement it. Maybe just saying "via the Into trait" or sth similar? Same below, with TryInto.

File: Scarb.toml
Comment (milancermak) [inline Scarb.toml:7]: Foundry should be a [`[dev-dependencies]`](https://docs.swmansion.com/scarb/docs/reference/specifying-dependencies.html#development-dependencies)

File: README.md
Comment (milancermak) [inline README.md:89]: The lib should be a Scarb package. Installation will then be just adding one line into `Scarb.toml` [dependencies] (and eventually, once Scarb has a package registry, even just `scarb add cairo-wadray`).

File: Scarb.toml
Comment (milancermak) [inline Scarb.toml:2]: Maybe just `wadray`? WDYT? IMO it's clear it's in Cairo.

**Review by milancermak — COMMENTED** [review]
IDK if you want to do it in this PR or a follow up one, but the library should be made into a Scarb package so it's easily installable. I think the bare minimum to do this is just adding a `[lib]` into Scarb.toml. But to be a bit more complete and helpful, it's worthwhile to add more of the [keys that can appear in the `[package]` section](https://docs.swmansion.com/scarb/docs/reference/manifest.html#package). For reference, here's [what I did](https://github.com/milancermak/xoroshiro-cairo/blob/master/Scarb.toml) in my xoroshiro lib a while ago.

I'm assuming support for Signed will also come eventually? 😇

File: README.md
Comment (tserg) [reply README.md:78]: I will change the phrasing across the README as well for clarity.

**Comment by tserg** [comment]
> I'm assuming support for Signed will also come eventually? 

Yes, I will add it in a separate PR. Should I move the tests into a new `tests/` directory here, or wait for the signed wadray PR?

File: README.md
Comment (tserg) [reply README.md]: Done! I will verify that the package installation works once this is merged.

**Comment by tserg** [comment]
> re: tests dir - idk, whatever works.

Okay, I will leave it to the signed PR to clean it up.

**Comment by tserg** [comment]
Going ahead to merge so I can check on the installation as Scarb package, and to unblock Signed PR.

**Comment by tserg** [comment]
I have verified that installing this library as a Scarb dependency works locally.

**Comment by milancermak** [comment]
Great stuff

**Pull request merged**
